optimum/onnxruntime/trainer.py

```diff
@@ -130,6 +131,54 @@
 SCALER_NAME = "scaler.pt"
 
 
+class ModuleWithLoss(nn.Module):
+    def __init__(self, model, args) -> None:
+        super().__init__()
+        self._original_model = model
+        self.args = args
+        # Label smoothing
+        if self.args.label_smoothing_factor != 0:
+            from transformers.trainer_pt_utils import LabelSmoother
+
+            self.label_smoother = LabelSmoother(epsilon=self.args.label_smoothing_factor)
+        else:
+            self.label_smoother = None
+
+    def forward(self, inputs: Dict[str, Union[torch.Tensor, Any]], return_outputs):
+        if self.label_smoother is not None and "labels" in inputs:
```

Comment:
The logic in the forward are mostly from transformer Trainer's compute_loss.

To avoid the duplicated code, can we bind that transformer Trainer's compute_loss function as member function of ModuleWithLoss, then we just call self.compute_loss(self._original_model ) to get the loss. This would make the maintenance easier I think.